### ISSUE TITLE
Wrong laser waist value when using LASY 

### ISSUE THREAD

**Issue by Gholipoorr** [fbpic/fbpic#673]
Hello
I try to use LASY for initiliazing laser pulse, but for some value of laser waist I got wrong value. I understand that for  fewer values than 10e-6 and more than 15e-6 this problem occurs. LASY and FBPIC scripts are attached.
[lwfa_script.txt](https://github.com/fbpic/fbpic/files/13687609/lwfa_script.txt)
[Profile.txt](https://github.com/fbpic/fbpic/files/13687611/Profile.txt)

**Comment by RemiLehe** [comment]
Thanks for raising this issue.

If I understand your question correctly, I think that you are saying that, in the above example, `ts.get_laser_waist` returned a different value than the one you expected (3 microns, i.e. the spot size that you used in `Profile.txt` file).

I think that the above observation is in fact expected, because of the line `laser.propagate(150.e-6)` in your `Profile.txt`. This line means that the laser is to be initialized 150 microns after its focal plane. Because the Rayleigh length here is shorter than 150 microns, the laser waist of the laser will be very different than 3 microns at initialization.

So, in short, I don't think that there is a bug here. If you remove `laser.propagate(150.e-6)`, you should now see the expected waist.

**Comment by Gholipoorr** [comment]
Thanks for your answer, I think  `laser. propagate` should be fewer than Rayleigh's length, is it true?

**Comment by RemiLehe** [comment]
Hm, in principle, `laser.propagate` can be used for distances longer than the Rayleigh length ; it all depends on what you want to simulate...
